packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/RealmSyncUtils.kt

```diff
@@ -69,10 +70,10 @@ internal fun <T, R> channelResultCallback(
 internal fun convertSyncError(error: SyncError): SyncException {
     // TODO In a normal environment we only expose the information in the SyncErrorCode.
     //  In debug mode we could consider using `detailedMessage` instead.
-    return convertSyncErrorCode(error.errorCode)
+    return convertSyncErrorCode(error.errorCode, error)
 }
 
-internal fun convertSyncErrorCode(syncError: SyncErrorCode): SyncException {
+internal fun convertSyncErrorCode(syncError: SyncErrorCode, error: SyncError? = null): SyncException {
```

Comment:
I think that I don't follow you, this weirdness in `convertSyncError` is because errors in `SyncSessionTransferCompletionCallback` are `SyncErrorCode` rather than `SyncError`. If we received an actual `SyncError` we would be able to have a common path.

As a solution I have created a wrapper based in `SyncError` to be used in `SyncSessionTransferCompletionCallback`. That way we can create a `SyncError` for a `SyncErrorCode` and have a common error handling path.

🤯